### PR TITLE
HPCC-16913 DropZone file checks should be moved, so that not hit by remote logical file copy.

### PR DESCRIPTION
**Pull request by AttilaVamos**
Move DropZone check from FileSprayer class to CDFUengine class

Move checkSourceTarget() function call from
CDistributedFileSystem::exportFile() and CDistributedFileSystem::import()
functions to corresponding part of CDFUengine:runWU()

Signed-off-by: Attila Vamos <attila.vamos@gmail.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [x] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

## Testing:

Tested manually with ECLWatch and Spray/despray related regression Suite tests with --pq

### CODE REVIEW

**Comment by richardkchapman** [comment]
@jakesmith Please review

File: dali/dfu/dfurun.cpp [hunk 1]
```diff
@@ -438,17 +438,101 @@ class CDFUengine: public CInterface, implements IDFUengine
     }
 
 
+    // DropZone check
+    void checkFilePath(RemoteFilename & filename)
+    {
+
```
Comment (jakesmith) [inline dali/dfu/dfurun.cpp]: trivial: extra blank line

File: dali/dfu/dfurun.cpp [hunk 2]
```diff
@@ -438,17 +438,101 @@ class CDFUengine: public CInterface, implements IDFUengine
...
+                        LOG(MCdebugInfo, unknownJob, "No matching drop zone path to file path: '%s'", pfilePath);
+                }
+                else
+                {
+                    SCMStringBuffer dropZoneName;
+                    dropZone->getName(dropZoneName);
+                    LOG(MCdebugInfo, unknownJob, "Drop zone path '%s' is %svisible in ECLWatch."
+                            , dropZoneName.str()
+                            , (dropZone->isECLWatchVisible() ? "" : "not ")
+                            );
```
Comment (jakesmith) [inline dali/dfu/dfurun.cpp]: minor: Probably the same as before, but this is very verbose, tracing every file checked, particularly verbose in the multi filename cases. I think should remove.

File: dali/dfu/dfurun.cpp [hunk 3]
```diff
@@ -438,17 +438,101 @@ class CDFUengine: public CInterface, implements IDFUengine
...
+                }
+
```
Comment (jakesmith) [inline dali/dfu/dfurun.cpp]: trivial: extra line

File: dali/dfu/dfurun.cpp [hunk 4]
```diff
@@ -580,8 +664,8 @@ class CDFUengine: public CInterface, implements IDFUengine
 
     bool performMonitor(IDFUWorkUnit *wu,IDFUmonitor *monitor,IConstDFUfileSpec *source, bool raiseexception, StringAttrArray *eventstriggered, StringAttrArray *eventsfile, IUserDescriptor *user)
     {
-        
-        
+
+
```
Comment (jakesmith) [inline dali/dfu/dfurun.cpp]: trivial: not new, but worth removing these empty lines.

**Comment by jakesmith** [comment]
@AttilaVamos - looks good, one comment, re. logging.

File: dali/dfu/dfurun.cpp (hunk 1 above)
Comment (AttilaVamos) [inline dali/dfu/dfurun.cpp]: Removed

File: dali/dfu/dfurun.cpp (hunk 2 above)
Comment (AttilaVamos) [inline dali/dfu/dfurun.cpp]: I would keep it in Debug build.

File: dali/dfu/dfurun.cpp (hunk 3 above)
Comment (AttilaVamos) [inline dali/dfu/dfurun.cpp]: Removed

File: dali/dfu/dfurun.cpp (hunk 4 above)
Comment (AttilaVamos) [inline dali/dfu/dfurun.cpp]: Removed

**Comment by AttilaVamos** [comment]
@jakesmith updated, please take a look.

File: dali/ft/filecopy.cpp [hunk 5]
```diff
@@ -3327,6 +3247,12 @@ void FileSprayer::updateTargetProperties()
             }
             curHistory->addPropTree("Origin",newRecord.getClear());
         }
+
+        int expireDays = options->getPropInt("@expireDays");
+        if (expireDays != -1)
+        {
+            curProps.setPropInt("@expireDays", expireDays);
+        }
```
Comment (jakesmith) [inline dali/ft/filecopy.cpp]: I can't see the difference between previous PR state and current (I assume squashed?)..
but I don't think this was here before and doesn't look related to this JIRA.

Comment (AttilaVamos) [inline dali/ft/filecopy.cpp]: Something went wrong.

**Comment by AttilaVamos** [comment]
@jakesmith problem (I hope) is fixed, check now.

**Comment by jakesmith** [comment]
@AttilaVamos - looks good.

@richardkchapman - ready to merge.

**Comment by AttilaVamos** [comment]
@richardkchapman please merge.

**Pull request merged**